src/gateways/useCreateGenericGatewayMutation.js

```diff
@@ -1,36 +1,10 @@
 import { useDataMutation } from '@dhis2/app-runtime'
-
-/**
- * This the shape of the additional fields
- * that can be added to a generic config
- *
- * @typedef {Object} Parameter
- * @prop {string} key
- * @prop {string} value
- * @prop {bool} header
- * @prop {bool} encode
- * @prop {bool} confidential
- */
-
-/**
- * @param {Object} args
- * @param {string} args.name
- * @param {string} args.urlTemplate
- * @param {Parameter[]} args.parameters
- *
- * @returns {Object}
- */
-const createDataFromVariables = ({ name, urlTemplate, parameters }) => ({
-    type: 'http',
-    name,
-    urlTemplate,
-    parameters,
-})
+import { createGenericGateWayDataFromVariables } from './createGenericGateWayDataFromVariables'
```

Comment:
Feels to me like abstracting this to a separate file makes it harder to understand what's happening. But doesn't make a functional difference so probably better addressed later when refactoring, if at all.